src/controllers/v1/entities.js

```diff
@@ -57,28 +59,403 @@ module.exports = class Entities extends Abstract {
   ]
 */
 
-  /**
-   * List all the entities.
-   * @method
-   * @name list 
-   * @returns {JSON} - List of all entities.
-   */
-
-  async list(req) {
-    return new Promise(async (resolve, reject) => {
-      try {
-
-        let result = await entitiesHelper.list("all", { name: 1 })
-
-        return resolve(result)
-
-      } catch (error) {
-        return reject({
-          status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
-          message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
-          errorObject: error
-        })
-      }
-    })
-  }
-}
\ No newline at end of file
+	/**
+	 * Find all the entities.
+	 * @method
+	 * @name find
+	 * @returns {JSON} - List of all entities.
+	 */
+
+	find(req) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				let entityData = await entitiesQueries.entityDocuments(req.body.query, req.body.projection)
+				return resolve({
+					success: true,
+					message: CONSTANTS.apiResponses.ASSETS_FETCHED_SUCCESSFULLY,
+					result: entityData,
+				})
+			} catch (error) {
+				return reject({
+					status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+					message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+					errorObject: error,
+				})
+			}
+		})
+	}
+
+	relatedEntities(req) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				let result = {}
+
+				let relatedEntities = await entitiesHelper.relatedEntities(req.params._id)
+				result['relatedEntities'] = relatedEntities.length > 0 ? relatedEntities : []
+
+				return resolve({
+					message: CONSTANTS.apiResponses.ENTITY_FETCHED,
+					result: result,
+				})
+			} catch (error) {
+				return reject({
+					status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+					message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+					errorObject: error,
+				})
+			}
+		})
+	}
+
+	mappingUpload(req) {
+		return new Promise(async (resolve, reject) => {
+			try {
+				let entityCSVData = await csv().fromString(req.files.entityMap.data.toString())
+
+				let entityMappingUploadResponse = await entitiesHelper.processEntityMappingUploadData(entityCSVData)
+				if (!entityMappingUploadResponse.success) {
+					throw new Error(CONSTANTS.apiResponses.SOMETHING_WENT_WRONG)
+				}
+
+				return resolve({
+					message: CONSTANTS.apiResponses.ENTITY_INFORMATION_UPDATE,
+					result: entityMappingUploadResponse,
+				})
+			} catch (error) {
+				return reject({
+					status: error.status || HTTP_STATUS_CODE.internal_server_error.status,
+					message: error.message || HTTP_STATUS_CODE.internal_server_error.message,
+					errorObject: error,
+				})
+			}
+		})
+	}
+
+	/**
+	 * details of the entities.
+	 * @method
+	 * @name details
+	 * @returns {JSON} - provide the details.
+	 */
+
+	details(req) {
```

Comment:
@MallanagoudaB api description is incomplete, add complete description. applicable to all cases like this